tests/components/tplink/test_sensor.py

```diff
@@ -252,10 +262,14 @@ async def test_sensor_children(
         child_entity = entity_registry.async_get(child_entity_id)
         assert child_entity
         assert child_entity.unique_id == f"PLUG{plug_id}DEVICEID_consumption_this_month"
-        assert child_entity.device_id != entity.device_id
         child_device = device_registry.async_get(child_entity.device_id)
         assert child_device
-        assert child_device.via_device_id == device.id
+        if on_parent_device:
+            assert child_entity.device_id == entity.device_id
+            assert child_device.connections == device.connections
+        else:
+            assert child_entity.device_id != entity.device_id
+            assert child_device.via_device_id == device.id
```

Comment:
Can we avoid the direct branching in the test?

Reply:
Now split into separate tests